File: X-Clone-Backend-FastAPI/app/models/follow.py
from __future__ import annotations

from sqlalchemy import Column, Integer

from ..db import Base


class Follow(Base):
    __tablename__ = "follows"

    id = Column(Integer, primary_key=True, autoincrement=True)
    follower_id = Column(Integer, nullable=False)
    followed_id = Column(Integer, nullable=False)

File: X-Clone-Backend/src/main/java/com/xclone/xclone/domain/follow/Follow.java
package com.xclone.xclone.domain.follow;

import jakarta.persistence.*;

@Entity
@Table(name = "follows")
public class Follow {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "followed_id")
    private Integer followedId;

    @Column(name = "follower_id")
    private Integer followerId;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getFollowedId() {
        return followedId;
    }

    public void setFollowedId(Integer followedId) {
        this.followedId = followedId;
    }

    public Integer getFollowerId() {
        return followerId;
    }

    public void setFollowerId(Integer followerId) {
        this.followerId = followerId;
    }
}
